src/Jobs/StructureOrders.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace RecursiveTree\Seat\TreeLib\Jobs;
+
+use Illuminate\Support\Facades\DB;
+use RecursiveTree\Seat\TreeLib\Models\MarketOrder;
+use Seat\Eseye\Exceptions\RequestFailedException;
+use Seat\Eveapi\Jobs\AbstractAuthCharacterJob;
+use Seat\Eveapi\Jobs\EsiBase;
+use Seat\Eveapi\Models\RefreshToken;
+use Seat\Eveapi\Models\Universe\UniverseStructure;
+
+/**
+ * Class Orders.
+ *
+ * @package Seat\Eveapi\Jobs\Market
+ */
+class StructureOrders extends AbstractAuthCharacterJob
+{
+
+    /**
+     * @var string
+     */
+    protected $method = 'get';
+
+    /**
+     * @var string
+     */
+    protected $endpoint = '/markets/structures/{structure_id}/';
+
+    /**
+     * @var string
+     */
+    protected $version = 'v1';
+
+    /**
+     * @var string
+     */
+    protected $scope = 'esi-markets.structure_markets.v1';
+
+    /**
+     * @var array
+     */
+    protected $tags = ['market', 'structure'];
+
+    /**
+     * @var int The structure ID to which this job is related.
+     */
+    protected $structure_id;
+
+    public function __construct(RefreshToken $token, UniverseStructure $structure)
+    {
+        $this->structure_id = $structure->structure_id;
+
+        parent::__construct($token);
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @throws \Throwable
+     */
+    public function handle()
+    {
+        $structure_id = $this->getStructureId();
+        $structure = UniverseStructure::find($structure_id);
+
+        MarketOrder::where('location_id', '=', $this->getStructureId())
+            ->delete();
+
+        //load all market data
+        while (true) {
+            //retrieve one page of market orders
+            try {
+                $orders = $this->retrieve(['structure_id' => $structure_id]);
+            } catch (RequestFailedException $e) {
+                dd($e);
```

Comment:
please remove dd() from production code. it is fine for debugging, but usually we just let it fail if it fails

Reply:
Got It I will remove